src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,134 @@
+                }
+                if (!line.contains("/at ") && command.equals(COMMAND_EVENT)) {
+                    throw new DukeException("☹ OOPS!!! Missing or incorrect /at statement");
+                }
+                return line.substring(line.indexOf(' ') + 1, line.indexOf('/') - 1);
+            default:
+                return null;
+            }
+        }
+    }
+
+    public static String getTaskDate(String command, String line) {
+        switch (command) {
+        case COMMAND_DEADLINE:
+            return line.split("/by ")[1];
+        case COMMAND_EVENT:
+            return line.split("/at ")[1];
+        default:
+            return null;
+        }
+    }
+
+    public static int getDoneIndex(String command, String line) throws DukeException {
+        if (line.split(command).length < 2 || line.split(command)[1].equals(" ")) {
+            throw new DukeException("☹ OOPS!!! Missing index of duke.task!");
+        }
+        try {
+            return Integer.parseInt(line.split(" ")[1]) - 1;
+        } catch (NumberFormatException e) {
+            throw new DukeException("☹ OOPS!!! No integer index detected!");
+        }
+    }
+
+    public static void main(String[] args) {
+        printGreetingMessage();
+
+        // Initialize new instance of a TaskList object
+        TaskList list = new TaskList();
+
+        // Initialize Scanner and read in user input
+        Scanner in = new Scanner(System.in);
+        String line = in.nextLine();
+        String command = getCommand(line);
+
+        while (!command.equals(COMMAND_BYE)) {
+            printHorizontalLine();
+            try {
+                switch (command) {
+                case COMMAND_LIST:
+                    list.printList();
+                    break;
+                case COMMAND_DONE:
+                    if (getDoneIndex(command, line) + 1 > list.getListLength()) {
+                        throw new DukeException("☹ OOPS!!! Index specified is out of list size!");
+                    }
+                    list.markAsDone(getDoneIndex(command, line));
+                    break;
+                case COMMAND_TODO:
+                    list.addToList(new ToDo(getTaskDescription(command, line)));
+                    break;
+                case COMMAND_DEADLINE:
+                    list.addToList(new Deadline(getTaskDescription(command, line), getTaskDate(command, line)));
+                    break;
+                case COMMAND_EVENT:
+                    list.addToList(new Event(getTaskDescription(command, line), getTaskDate(command, line)));
+                    break;
+                default:
+                    throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means!");
+                }
+            } catch (DukeException e) {
+                System.out.println(e.toString());
+            }
+
+            printHorizontalLine();
+
+            // Read next line
+            line = in.nextLine();
```

Comment:
Perhaps can extract as method for these two lines, to better align with SLAP paradigm?